### ISSUE TITLE
Add ability to use icons with Tabs

### ISSUE THREAD

**Issue by SpyTec** [patternfly/patternfly-react#2536]
Right now in PF4, you cannot add PF icons to Tab components. This issue is a feature request to add that functionality

My first thought is having a new prop for icon where the React component is sent in

**Comment by rachael-phillips** [comment]
Hi @SpyTec thanks for filing this! @mcarrano is this something we ever talked about in the design process or would recommend?

**Comment by mcarrano** [comment]
@rachael-phillips we had not considered an option to add icons to tabs. @SpyTec can you give me a bit more background about your usage and why you need this?

**Comment by SpyTec** [comment]
As an example of what I mean:

![image](https://user-images.githubusercontent.com/6598829/61464280-4d528700-a976-11e9-9708-10b681065750.png)

I could just see this as better UX as you can at a glance see what each tab is, such as you would on a navigation. On smaller devices or screens one could even reduce the overall size by only showing icon

![image](https://user-images.githubusercontent.com/6598829/61464534-b1754b00-a976-11e9-9283-81aa98c36aa6.png)

**Comment by kybaker** [comment]
@mcarrano My only concern with this proposed solution is that it maintains accessibility when only showing an icon in the smallest view port size.

**Comment by mcarrano** [comment]
@SpyTec @rachael-phillips We currently have some work planning to rework our tab designs here https://github.com/patternfly/patternfly-next/issues/1855  We can factor this in.

**Comment by karelhala** [comment]
No need to add new prop to tab component. You should be able to pass any component to title of tab
`<Tab eventKey={0} title={<div><UserFriendsIcon/>Users</div>}>` however the prop is wrongly defined as it states that you can pass sring only and it will throw error in console because of this.

**Comment by karelhala** [comment]
> Speaking of different resolutions, I would do something want something like this as well

This doesn't really have to be part of PF core styles, it is nice to have but not something critical. I would assume that if you want such thing you'd have to write styles for this. Because you'd also have to provide accessibility attributes and such.

**Comment by SpyTec** [comment]
> No need to add new prop to tab component. You should be able to pass any component to title of tab
> `<Tab eventKey={0} title={<div><UserFriendsIcon/>Users</div>}>` however the prop is wrongly defined as it states that you can pass sring only and it will throw error in console because of this.

We can't simply update the type definitions however since TabProps extends React.HTMLProps. We'd then get this error:
```
Interface 'TabProps' incorrectly extends interface 'HTMLProps<HTMLAnchorElement | HTMLButtonElement>'.
  Types of property 'title' are incompatible.
    Type 'ReactNode' is not assignable to type 'string'.
```

**Comment by karelhala** [comment]
@SpyTec that's why I opened this PR https://github.com/patternfly/patternfly-react/pull/2766 to change prop type of title to `React.ReactNode` (string, number, function).

**Comment by SpyTec** [comment]
Fixed in #2681
